### ISSUE TITLE
Implement upsert update option

### ISSUE THREAD

**Issue by YurySolovyov** [YuriSolovyov/Dexie-mongoify#3]
Update method should have a following signature [as per mongo native driver](http://mongodb.github.io/node-mongodb-native/2.1/api/Collection.html#update): 

``` javascript
.update(query, updates[, options]); // -> Promise
```

Let's implement `upsert` option first.

**Comment by YurySolovyov** [comment]
@nevf, I started working on results normalization in `result` branch, is there any particular result properties you are interested in?
ATM result looks like so:

``` javascript
{
    result: {
        ok: 1,
        nModified: modified
    },

    modifiedCount: modified,
    upsertedCount: upserted,
    upsertedId: id
};
```

**Comment by nevf** [comment]
@YuriSolovyov That looks good. It is more about being as compatible as possible with the Mongo Native driver, so code that works with it will work with the mongoify, unchanged or with minimal changes.

**Comment by YurySolovyov** [comment]
This is landed as #4, results shape normalization can be tracked in #8
